components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java

```diff
@@ -349,7 +358,42 @@ private AdvancedApplicationConfiguration buildAdvancedAppConfiguration(ServicePr
                 .skipLoginConsent(authConfig.isSkipConsent())
                 .skipLogoutConsent(authConfig.isSkipLogoutConsent())
                 .certificate(getCertificate(serviceProvider))
-                .fragment(isFragmentApp(serviceProvider));
+                .fragment(isFragmentApp(serviceProvider))
+                .additionalSpProperties(getSpProperties(serviceProvider));
+    }
+
+    private List<AdditionalSpProperty> getSpProperties(ServiceProvider serviceProvider) {
+
+        ServiceProviderProperty[] serviceProviderProperties = serviceProvider.getSpProperties();
+        ServiceProviderProperty[] updatedSpProperties = removeAndSetSpProperties(serviceProviderProperties);
+        List<AdditionalSpProperty> additionalSpProperties = new ArrayList<>();
+        for (ServiceProviderProperty serviceProviderProperty : updatedSpProperties) {
+            AdditionalSpProperty spProperties = new AdditionalSpProperty();
+            if (StringUtils.isNotBlank(serviceProviderProperty.getName())) {
+                spProperties.setName(serviceProviderProperty.getName());
+                spProperties.setValue(serviceProviderProperty.getValue());
+            }
+            if (StringUtils.isNotBlank(serviceProviderProperty.getDisplayName())) {
+                spProperties.setDisplayName(serviceProviderProperty.getDisplayName());
+            }
+            additionalSpProperties.add(spProperties);
+        }
+        return additionalSpProperties;
+    }
+
+    private ServiceProviderProperty[] removeAndSetSpProperties(ServiceProviderProperty[] propertyList) {
+
+        /* These properties are part of advanced configurations and hence removing
+        them as they can't be packed as a part of additional sp properties again.*/
+            List<ServiceProviderProperty> spPropertyList =
+                    Arrays.stream(propertyList).collect(Collectors.toList());
+            spPropertyList.removeIf(property -> SKIP_CONSENT.equals(property.getName()));
+            spPropertyList.removeIf(property -> SKIP_LOGOUT_CONSENT.equals(property.getName()));
+            spPropertyList.removeIf(property -> USE_DOMAIN_IN_ROLES.equals(property.getName()));
+            spPropertyList.removeIf(property -> USE_USER_ID_FOR_DEFAULT_SUBJECT.equals(property.getName()));
+            spPropertyList.removeIf(property -> TEMPLATE_ID_SP_PROPERTY_NAME.equals(property.getName()));
+            spPropertyList.removeIf(property -> IS_MANAGEMENT_APP_SP_PROPERTY_NAME.equals(property.getName()));
+            return spPropertyList.toArray(new ServiceProviderProperty[0]);
```

Comment:
Possible duplicate ? https://github.com/wso2/identity-api-server/pull/366/files#diff-f8e31673cfd2b9fbe65a60eccec74a73e57e53f5154b2d85b7b530e3fdd74d87R88-R99

Reply:
So, suggesting to move it to as a util method? The data types have a difference though (ServiceProviderProperty vs SpAdditionalProperty)